cmd/system-probe/modules/dynamic_instrumentation.go

```diff
@@ -14,21 +14,22 @@ import (
 	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
 	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
 	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
-	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation"
+	dimod "github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/module"
 	"github.com/DataDog/datadog-agent/pkg/ebpf"
 )
 
-// DynamicInstrumentation is the dynamic instrumentation module factory
+// DynamicInstrumentation is a system probe module which allows you to add instrumentation into
+// running Go services without restarts.
 var DynamicInstrumentation = module.Factory{
 	Name:             config.DynamicInstrumentationModule,
 	ConfigNamespaces: []string{},
 	Fn: func(agentConfiguration *sysconfigtypes.Config, _ module.FactoryDependencies) (module.Module, error) {
-		config, err := dynamicinstrumentation.NewConfig(agentConfiguration)
+		config, err := dimod.NewConfig(agentConfiguration)
 		if err != nil {
 			return nil, fmt.Errorf("invalid dynamic instrumentation module configuration: %w", err)
 		}
 
-		m, err := dynamicinstrumentation.NewModule(config)
+		m, err := dimod.NewModule(config)
 		if errors.Is(err, ebpf.ErrNotImplemented) {
 			return nil, module.ErrNotEnabled
 		}
```

Comment:
Where is the logic that checks for `DD_DYNAMIC_INSTRUMENTATION_ENABLED` for datadog-agent?

I saw where we are checking the env var on the target process, but I can't find where the agent config value is used.

Reply:
On this line the env var is bound to the config value: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/setup/system_probe.go#L169

Reply:
And it being enabled is checked here: https://github.com/DataDog/datadog-agent/blob/main/cmd/system-probe/api/module/loader.go#L69